server/handlers.go

```diff
@@ -668,6 +668,54 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 			s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
 			return
 		}
+
+		tokenRef := storage.RefreshTokenRef{
+			ID:        refresh.ID,
+			ClientID:  refresh.ClientID,
+			CreatedAt: refresh.CreatedAt,
+			LastUsed:  refresh.LastUsed,
+		}
+
+		if session, err := s.storage.GetOfflineSessions(refresh.Claims.UserID, refresh.ConnectorID); err != nil {
+			if err == storage.ErrNotFound {
+				offlineSessions := storage.OfflineSessions{
+					UserID:  refresh.Claims.UserID,
+					ConnID:  refresh.ConnectorID,
+					Refresh: make(map[string]*storage.RefreshTokenRef),
+				}
+				offlineSessions.Refresh[tokenRef.ClientID] = &tokenRef
+
+				if err := s.storage.CreateOfflineSessions(offlineSessions); err != nil {
+					s.logger.Errorf("failed to create offline session: %v", err)
+					s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
+					return
+				}
+			} else {
+				s.logger.Errorf("failed to get offline session: %v", err)
+				s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
+				return
+			}
+		} else {
+			if oldTokenRef, ok := session.Refresh[tokenRef.ClientID]; ok {
```

Comment:
Tested to see that this works fine. It gets interpreted as an empty map after its unmarshaled. Also, once we add token revocation, if there is ever a case when the `OfflineSession` object has an empty `Refresh` map, we would clean up the entire `OfflineSession` object.